core/src/main/java/com/amplifyframework/analytics/UserProfile.java

```diff
@@ -0,0 +1,314 @@
+        if (email != null ? !email.equals(that.email) : that.email != null) return false;
+        if (plan != null ? !plan.equals(that.plan) : that.plan != null) return false;
+        if (location != null ? !location.equals(that.location) : that.location != null)
+            return false;
+        return customProperties != null ? customProperties.equals(that.customProperties) :
+                that.customProperties == null;
+    }
+
+    @SuppressWarnings("checkstyle:MagicNumber")
+    @Override
+    public int hashCode() {
+        int result = name != null ? name.hashCode() : 0;
+        result = 31 * result + (email != null ? email.hashCode() : 0);
+        result = 31 * result + (plan != null ? plan.hashCode() : 0);
+        result = 31 * result + (location != null ? location.hashCode() : 0);
+        result = 31 * result + (customProperties != null ? customProperties.hashCode() : 0);
+        return result;
+    }
+
+    @NonNull
+    @Override
+    public String toString() {
+        return "UserProfile{" +
+                "name='" + name + '\'' +
+                ", email='" + email + '\'' +
+                ", plan='" + plan + '\'' +
+                ", location=" + location +
+                ", customProperties=" + customProperties +
+                '}';
+    }
+
+    public static final class Builder {
+        private String name;
+        private String email;
+        private String plan;
+        private Location location;
+        private Properties customProperties;
+
+        @NonNull
+        public Builder name(@NonNull final String name) {
+            Objects.requireNonNull(name);
+            this.name = name;
+            return this;
+        }
+
+        @NonNull
+        public Builder email(@NonNull final String email) {
+            Objects.requireNonNull(email);
+            this.email = email;
+            return this;
+        }
+
+        @NonNull
+        public Builder plan(@NonNull final String plan) {
+            Objects.requireNonNull(plan);
+            this.plan = plan;
+            return this;
+        }
+
+        @NonNull
+        public Builder location(@NonNull final Location location) {
+            Objects.requireNonNull(location);
+            this.location = location;
+            return this;
+        }
+
+        @NonNull
+        public Builder customProperties(@NonNull final Properties properties) {
+            Objects.requireNonNull(properties);
+            this.customProperties = properties;
+            return this;
+        }
+
+        @NonNull
+        public UserProfile build() {
+            return new UserProfile(this);
+        }
+    }
+
+    public static final class Location {
+        private final Double latitude;
+        private final Double longitude;
+        private final String postalCode;
+        private final String city;
+        private final String region;
+        private final String country;
+
+        public Location(@NonNull Builder builder) {
```

Comment:
Same deal here, you can probably make this `private Location(Builder builder)`, and then as long as you always pass non-null from `build()`, you don't need to check `Objects.requireNonNull(...)`, since that's the only single place it'll ever be possible.